sw/device/lib/dif/dif_entropy.h

```diff
@@ -682,6 +699,24 @@ dif_entropy_result_t dif_entropy_fifo_set_capacity(const dif_entropy_t *entropy,
 DIF_WARN_UNUSED_RESULT
 dif_entropy_result_t dif_entropy_fifo_reconnect(const dif_entropy_t *entropy);
 
+/**
+ * Disables the entropy module
+ *
+ * @param entropy An entropy source handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_disable(const dif_entropy_t *entropy);
+
+/**
+ * Get main entropy fsm idle status
+ *
+ * @param entropy An entropy source handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_get_idle(const dif_entropy_t *entropy);
```

Comment:
Why doesn't this return a bool out-param for the idleness state? (also, it should probably be called `is_idle`...

Reply:
yeah i did it this way because then i could just do `while(get_idle() != Ok`.  Is that not the recommended approach?
Since i needed to poll on it, it seemed less cumbersome than having to pull the variable out to check.

Reply:
ah i see how this is done differently in modules like aes now, i can convert to the same.